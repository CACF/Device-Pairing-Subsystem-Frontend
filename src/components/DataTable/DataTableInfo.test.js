import React from 'react';
import {shallow, mount} from 'enzyme';
import DataTableInfo from './DataTableInfo';

const pageInfo = {
  start: 1,
  limit: 10,
  total: 50,
  itemType: 'requests'
}

describe('DataTableInfo', () => {

  /* Test if render correctly */
  test('should render correctly', () => {
    const wrapper = shallow(<DataTableInfo />);
    expect(wrapper).toMatchSnapshot();
  });

  /* Test if DataTableInfo renders successfully */
  test('should render DataTableInfo successfully', () => {
    const wrapper = shallow(<DataTableInfo />);
    expect(wrapper.exists()).toBe(true);
  });

  /* Test DataTableInfo have certain props */
  test('DataTableInfo have certain props', () => {
    const wrapper = mount(<DataTableInfo start={pageInfo.start} limit={pageInfo.limit} total={pageInfo.total} itemType={pageInfo.itemType} />);
    expect(wrapper.find('DataTableInfo').props().start).toBe(1);
    expect(wrapper.find('DataTableInfo').props().limit).toBe(10);
    expect(wrapper.find('DataTableInfo').props().total).toEqual(50);
    expect(wrapper.find('DataTableInfo').props().itemType).toEqual('requests');
  });

  /* Test DataTableInfo wrapper has specific class */
  test('DataTableInfo wrapper has specific class and spans', () => {
    const wrapper = mount(<DataTableInfo start={pageInfo.start} limit={pageInfo.limit} total={pageInfo.total} itemType={pageInfo.itemType} />);
    expect(wrapper.find('p').hasClass('page-result')).toBe(true);
    expect(wrapper.find('p').find('span')).toHaveLength(2);
  });

  /* Test total should be used if pagination reaches to end */
  test('total requests count if pagination reaches to end', () => {
    const pageInfo = {
      start: 41,
      limit: 10,
      total: 47,
      itemType: 'requests'
    }
    const wrapper = mount(<DataTableInfo start={pageInfo.start} limit={pageInfo.limit} total={pageInfo.total} itemType={pageInfo.itemType} />);
    expect(wrapper.find('span').at(1).text()).toBe((pageInfo.total).toString());
  });

  /* Test requests limit should be test */
  test('request limit should be used', () => {
    const pageInfo = {
      start: 0,
      limit: 10,
      total: 47,
      itemType: 'requests'
    }
    const wrapper = mount(<DataTableInfo start={pageInfo.start} limit={pageInfo.limit} total={pageInfo.total} itemType={pageInfo.itemType} />);
    expect(wrapper.find('span').at(0).text()).toBe((pageInfo.start + 1) + ' to ' + (pageInfo.limit).toString());
  });

});