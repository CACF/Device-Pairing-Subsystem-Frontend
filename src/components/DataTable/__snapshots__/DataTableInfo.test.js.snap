// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableInfo should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataTableInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Showing ",
        <span>
          NaN
           to 
          NaN
        </span>,
        " of ",
        <span />,
        " ",
        undefined,
      ],
      "className": "page-result",
    },
    "ref": null,
    "rendered": Array [
      "Showing ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            NaN,
            " to ",
            NaN,
          ],
        },
        "ref": null,
        "rendered": Array [
          NaN,
          " to ",
          NaN,
        ],
        "type": "span",
      },
      " of ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      " ",
      undefined,
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Showing ",
          <span>
            NaN
             to 
            NaN
          </span>,
          " of ",
          <span />,
          " ",
          undefined,
        ],
        "className": "page-result",
      },
      "ref": null,
      "rendered": Array [
        "Showing ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              NaN,
              " to ",
              NaN,
            ],
          },
          "ref": null,
          "rendered": Array [
            NaN,
            " to ",
            NaN,
          ],
          "type": "span",
        },
        " of ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        " ",
        undefined,
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
